### PR TITLE
Ability to change the audio/video playback speed 

### PR DESCRIPTION
**Pull request by AnKh99**
I will be glad to comments

### CODE REVIEW

**Comment by cjmaxik** [comment]
Sorry for the lack of feedback, will try to look at it this week.

**Comment by cjmaxik** [comment]
Thank you for this feature. 
This PR will stay for now, since I have plans to add video support as well.

**Review by cjmaxik — APPROVED** [review]

File: source/_locales/en/messages.json [hunk 1]
```diff
@@ -101,5 +101,11 @@
   },
   "video_quality_tiny": {
     "message": "144p"
+  },
+  "options_force_audio_playback_rate": {
+    "message": "Change audio playback rate"
+  },
+  "options_audio_playback_rate": {
+    "message": "Playback rate"
```
Comment (cjmaxik) [inline source/_locales/en/messages.json:109]: Remove that

File: source/_locales/ru/messages.json [hunk 2]
```diff
@@ -77,5 +77,11 @@
   },
   "download_video_modal_title": {
     "message": "Выбери качество видео"
+  },
+  "options_force_audio_playback_rate": {
+    "message": "Изменять скорость воспроизведения аудио"
+  },
+  "options_audio_playback_rate": {
+    "message": "Скорость аудио"
```
Comment (cjmaxik) [inline source/_locales/ru/messages.json:85]: Remove that

File: source/content/playerChanges.js [hunk 3]
```diff
@@ -63,19 +63,80 @@ export const prepareAudioPlayer = async (element, options) => {
...
-    playerTitle.insertAdjacentHTML('afterEnd', templates.audioDownloadButton(audioUrl))
-    const link = element.querySelector('button.MB_audio_download')
-
+    const playbackRate = options.force_audio_playback_rate ? options.audio_playback_rate : 1.0
+    const buttonsWrapper = document.createElement('div')
+    buttonsWrapper.style.display = 'flex'
+    buttonsWrapper.style.justifyContent = 'center'
+    buttonsWrapper.style.marginLeft = 'auto'
+    buttonsWrapper.innerHTML = templates.audioSpeedController(playbackRate) + templates.audioDownloadButton(audioUrl)
+    audio.playbackRate = playbackRate
```
Comment (cjmaxik) [inline source/content/playerChanges.js:78]: Move to templates

File: source/content/playerChanges.js [hunk 4]
```diff
@@ -63,19 +63,80 @@ export const prepareAudioPlayer = async (element, options) => {
...
+      const currentPlaybackRateElements = document.querySelectorAll('span.Current_Playback_Rate')
+      currentPlaybackRateElements.forEach((element) => {
+        element.textContent = `x${audio.playbackRate}`
+      })
+
+      const currentAudioElements = document.querySelectorAll('audio')
+      currentAudioElements.forEach((element) => {
+        element.playbackRate = audio.playbackRate
+      })
+    })
```
Comment (cjmaxik) [inline source/content/playerChanges.js:121]: Duplicated code, please extract it to a separate function

File: source/global/options-storage.js [hunk 5]
```diff
@@ -6,7 +6,9 @@ export default new OptionsSync({
     force_video_quality: true,
     video_quality: '1080p',
     save_last_timestamp: true,
-    theater_mode: true
+    theater_mode: true,
+    force_audio_playback_rate: false,
```
Comment (cjmaxik) [inline source/global/options-storage.js:10]: Remove. `audio_playback_rate` is the only thing we care about.

File: source/options/options.js [hunk 6]
```diff
@@ -18,6 +18,25 @@ const videoQuality = [
...
+  '2',
+  '2.25',
+  '2.5',
+  '2.75',
+  '3',
+  '3.25',
+  '3.5',
+  '3.75',
+  '4'
+]
```
Comment (cjmaxik) [inline source/options/options.js:38]: Remove.

File: source/options/options.js [hunk 7]
```diff
@@ -69,6 +88,33 @@ const generateVideoQualityOptions = (options) => {
...
+  audioRateOptions.addEventListener('change', (event) => {
+    event.preventDefault()
+    optionsStorage.set({ audio_playback_rate: audioRateOptions.value })
+  })
+
+  forceAudioRate.addEventListener('change', (event) => {
+    audioRateOptions.disabled = !event.currentTarget.checked
+  })
+}
+
```
Comment (cjmaxik) [inline source/options/options.js:117]: Remove.

File: source/options/options.js [hunk 8]
```diff
@@ -79,6 +125,7 @@ const init = async () => {
   console.log(options)
 
   generateVideoQualityOptions(options)
+  generateAudioRateOptions(options)
```
Comment (cjmaxik) [inline source/options/options.js:128]: Remove.

**Review by cjmaxik — CHANGES_REQUESTED** [review]
Sorry for the very long absence. I've reviewed the changes. 
In short, it should only store/retrieve last audio speed used. There is no need to add them to the Settings page.

Also, please consider adding this feature for videos.

**Comment by AnKh99** [comment]
Hi, I'll update it. Thanks for comments

File: source/_locales/en/messages.json (hunk 1 above)
Comment (AnKh99) [reply source/_locales/en/messages.json]: done

File: source/_locales/ru/messages.json (hunk 2 above)
Comment (AnKh99) [reply source/_locales/ru/messages.json]: done

File: source/content/playerChanges.js (hunk 3 above)
Comment (AnKh99) [reply source/content/playerChanges.js]: done

File: source/content/playerChanges.js (hunk 4 above)
Comment (AnKh99) [reply source/content/playerChanges.js]: done

**Comment by AnKh99** [comment]
Please let me know if I pushed it incorrectly, I have never worked with github with someone else's project before)

**Comment by cjmaxik** [comment]
Looks good, thank you. 
I will make an attempt to port these changes to a video player.

**Comment by cjmaxik** [comment]
Added support for video + did a refactor. Please review and test the changes.

**Review by cjmaxik — APPROVED** [review]

**Pull request merged**